websocket.go

```diff
@@ -160,38 +170,41 @@ func (t *WebsocketTransport) Listen(a ma.Multiaddr) (tpt.Listener, error) {
 		return nil, err
 	}
 
-	tlist := t.wrapListener(list)
-
-	u, err := url.Parse("ws://" + list.Addr().String())
+	u, err := url.Parse("http://" + list.Addr().String())
 	if err != nil {
 		return nil, err
 	}
 
-	s := &ws.Server{
-		Handler: tlist.handleWsConn,
-		Config:  ws.Config{Origin: u},
-	}
+	tlist := t.wrapListener(list, u)
 
-	go http.Serve(list.NetListener(), s)
+	http.HandleFunc("/", tlist.handleWsConn)
```

Comment:
this is using the global http muxer, I would rename handleWsConn to `ServeHTTP(ResponseWriter, *Request)` and pass `tlist` instead of `nil` to `Serve` below.